views/scan.ejs

```diff
@@ -109,6 +109,31 @@
       </datalist>
       <table class="table table-bordered">
         <tbody>
+          <tr>
+            <th>
+              <select name="operationType">
+                <option value="scan"<% if (query.operationType === 'scan') { %> selected<% } %>>Scan</option>
+                <option value="query"<% if (query.operationType === 'query') { %> selected<% } %>>Query</option>
```

Comment:
I've selected `query` operation on my table with no `GlobalSecondaryIndexes` and pressed "scan" and got error response. As an occasional user of DynamoDB, I don't exactly know how things should work and what's missing so it would be nice to have some error handling for this. For example, disable "scan" button if selected options make no sense. Or even disable `query` operation completely if possible to determine that it will not work.

Reply:
You can query your main index using the `KeySchema` fields. I've made the interface automatically add the required fields when switching to the `query` operation now so that should help prevent common user errors.